vendor/bat-native-ads/src/bat/ads/internal/covariates/log_entries/notification_ad_event_unittest.cc

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/covariates/log_entries/notification_ad_event.h"
+
+#include <sstream>
+
+#include "bat/ads/confirmation_type.h"
+#include "bat/ads/internal/base/unittest/unittest_base.h"
+
+// npm run test -- brave_unit_tests --filter=BatAdsFederatedLogEntries*
+
+namespace {
+
+std::string GetValueForEventType(
+    const ads::mojom::NotificationAdEventType event_type) {
+  ads::NotificationAdEvent notification_ad_event;
+  notification_ad_event.SetEventType(event_type);
+  return notification_ad_event.GetValue();
+}
+
+std::string GetEventTypeAsString(
+    const ads::mojom::NotificationAdEventType event_type) {
+  std::stringstream ss;
+  ss << event_type;
+
+  return ss.str();
+}
+
+}  // namespace
+
+namespace ads {
+
+TEST(BatAdsFederatedLogEntriesNotificationAdEventTest, GetDataType) {
+  NotificationAdEvent notification_ad_event;
+
+  // Act
+  brave_federated::mojom::DataType data_type =
+      notification_ad_event.GetDataType();
+
+  // Assert
+  EXPECT_EQ(brave_federated::mojom::DataType::kString, data_type);
+}
+
+TEST(BatAdsFederatedLogEntriesNotificationAdEventTest, GetValueWhenClicked) {
+  // Arrange
+  NotificationAdEvent notification_ad_event;
+  mojom::NotificationAdEventType clicked_event_type =
+      mojom::NotificationAdEventType::kClicked;
+
+  // Act
+  const std::string value = GetValueForEventType(clicked_event_type);
+
+  // Assert
+  const std::string expected_value = GetEventTypeAsString(clicked_event_type);
+  EXPECT_EQ(expected_value, value);
+}
+
+TEST(BatAdsFederatedLogEntriesNotificationAdEventTest, GetValueWhenDismissed) {
+  // Arrange
+  NotificationAdEvent notification_ad_event;
+  mojom::NotificationAdEventType dismissed_event_type =
+      mojom::NotificationAdEventType::kDismissed;
+
+  // Act
+  const std::string value = GetValueForEventType(dismissed_event_type);
+
+  // Assert
+  const std::string expected_value = GetEventTypeAsString(dismissed_event_type);
+  EXPECT_EQ(expected_value, value);
+}
+
+TEST(BatAdsFederatedLogEntriesNotificationAdEventTest, GetValueWhenTimeout) {
+  // Arrange
+  NotificationAdEvent notification_ad_event;
+  mojom::NotificationAdEventType timed_out_event_type =
+      mojom::NotificationAdEventType::kTimedOut;
+
+  // Act
+  const std::string value = GetValueForEventType(timed_out_event_type);
```

Comment:
nit: use `mojom::NotificationAdEventType::kTimedOut` directly instead of assigning to a local variable